tensorboard/webapp/hparams/_redux/hparams_selectors.ts

```diff
@@ -178,3 +184,51 @@ export const getExperimentsHparamsAndMetricsSpecs = createSelector(
     );
   }
 );
+
+export const getDashboardHparamsAndMetricsSpecs = createSelector(
+  getHparamsState,
+  (state: HparamsState) => {
+    return state.dashboardSpecs;
+  }
+);
+
+export const getDashboardRunsToHparamsAndMetrics = createSelector(
+  getHparamsState,
+  (state): RunToHparamsAndMetrics => {
+    const runToHparamsAndMetrics: RunToHparamsAndMetrics = {};
+
+    for (const sessionGroup of state.dashboardSessionGroups) {
+      const hparams: HparamValue[] = Object.entries(sessionGroup.hparams).map(
+        (keyValue) => {
+          const [hparam, value] = keyValue;
+          return {name: hparam, value};
+        }
+      );
+
+      for (const session of sessionGroup.sessions) {
+        runToHparamsAndMetrics[session.name] = {
+          metrics: [],
+          hparams,
+        };
+
+        for (const metricValue of session.metricValues) {
```

Comment:
I recall in a recent discussion that we decided we didn't want metrics from the /session_groups response to appear in the Time Series runs table. You would instead be using the metric values from elsewhere in the data model - from the /timeSeries response perhaps?

We thought that we could therefore turn off metric calculation for /session_groups requests for the dashboard. We'd save a bunch of unneccesary processing and reduce latency.

With that in mind, is there any need for parsing the session metric values?

Reply:
From an offline discussion:
 * We will stop relying on this portion of the response for data.
 * We should calculate these values if they exist in the store particularly considering this is essentially the same as the logic found in the runs data source.